hack/version.py

```diff
@@ -31,142 +25,142 @@
 
 
 def parse_branch_name(branch_name):
-    """Split up a branch name of the form 'ocm-X.Y[-mce-M.N].
+    """Split up a branch name of the form 'mce-X.Y.
 
-    :param branch_name: A branch name. If of the form [remote/]ocm-X.Y[-mce-M.N] we will parse
+    :param branch_name: A branch name. If of the form [remote/]mce-M.N we will parse
         it as noted below; otherwise the first return will be False.
     :return parsed (bool): True if the branch_name was parseable; False otherwise.
     :return remote (str): If parsed and the branch_name contained a remote/ prefix, it is
         returned here; otherwise this is the empty string.
     :return prefix (str): Two-digit semver prefix of the bundle to be generated. If the branch
-        name is of the form [remote/]ocm-X.Y, this will be X.Y; if of the form
-        [remote/]ocm-X.Y-mce-M.N it will be M.N. If not parseable, it will be the empty string.
-    :return channel (str): The name of the channel in which we'll include the bundle. If the
-        branch name is of the form [remote/]ocm-X.Y, this will be ocm-X.Y; if of the form
-        [remote/]ocm-X.Y-mce-M.N it will be mce-M.N. If not parseable, it will be the empty
-        string.
+        name is of the form [remote/]mce-X.Y, this will be X.Y;
+        If not parseable, it will be the empty string.
     """
+
     m = MCE_BRANCH_RE.match(branch_name)
     if m:
-        return True, m.group(1), m.group(2), m.group(3)
-    m = OCM_BRANCH_RE.match(branch_name)
+        return True, m.group(1), m.group(2)
+    m = MASTER_BRANCH_RE.match(branch_name)
     if m:
-        return True, m.group(1), m.group(3), m.group(2)
-    return False, '', '', ''
+        return True, m.group(1), HIVE_VERSION_PREFIX
+    return False, "", ""
+
 
+def origin_branch_info(hive_repo):
+    """Find remote references in origin/ repository
 
-def remote_branch_info(hive_repo):
-    """"""
+    :param hive_repo: The git.Repo for the local hive clone.
+    :return ret (dict): A dictionary of all remote refs that match the "master" and "mce-*" format.
+    The key is the commit SHA for that ref and values are the prefix and the ref name."""
     ret = {}
     for ref in hive_repo.refs:
-        parsed, remote, prefix, channel = parse_branch_name(ref.name)
+        parsed, remote, prefix = parse_branch_name(ref.name)
         # Only pay attention to remote refs
         if not parsed:
             continue
         if remote != "origin/":
             continue
-        ret[ref.name] = dict(
-            prefix=prefix,
-            channel=channel,
-            commit_hash=hive_repo.rev_parse(ref.name).hexsha,
-        )
+        sha = hive_repo.rev_parse(ref.name).hexsha
+        if sha not in ret:
+            ret[sha] = dict(
+                prefix={prefix},
+                name={ref.name},
+            )
+        else:
+            ret[sha]["prefix"].add(prefix)
+            ret[sha]["name"].add(ref.name)
```

Comment:
jfyi, [collections.defaultdict](https://docs.python.org/3/library/collections.html#collections.defaultdict) exists for this use case